examples/dynamic-backend/main.tf

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+provider "google" {
+  project = var.project
+}
+
+provider "google-beta" {
+  project = var.project
+}
+
+resource "google_compute_managed_ssl_certificate" "frontend" {
+  name = "managedcert"
+
+  managed {
+    domains = var.domains
+  }
+}
+
+module "load_balancer" {
+  source = "../../modules/dynamic_backends"
```

Comment:
I have not tested this, but we use something similar at Prefect

Reply:
Thanks we can look into adding one as a follow up